### DISCUSSION TITLE
Comparator Library Questions

### DISCUSSION THREAD

**Discussion by stevecolt** [SpenceKonde/megaTinyCore#472] [Library discussion]
Hello,
The answer may be blindingly obvious  (when you know!!), but how do you read the output pin in the Comparator library? I am using a 1614 part and using 2 comparators basically as level shifters.
The internal reference example works fine, and I can see the outputs toggling happily on my scope, the interrupt example also works fine, and I can see interrupts happening with the output disabled.
What I cant figure out is how to read the data on a none interrupt line. Ideally I would like to disable the output on this channel as well, and read the comparator status bit (bit4) instead , or any other method( pins are scarce on 1614!!!).
I feel this is really something that should be in the library,, but Ive no idea how to do it. If I HAVE to use pin how do I read it? DigitalRead(ACOut) doesn't seem to work. Any help please
Thanks 
Steve

**Comment by grandaspanna** [comment]
Possible suggestion: check the STATE bit of the AC STATUS register (AC0.STATUS, I think)

**Reply by SpenceKonde** [reply]
Something like that is possible, that's what Comparator.read() does, but the whole point of those wrapper libraries is precisely so you don't have to do that. The fact that the functionality was undocumented is very unfortunate!)

**Comment by SpenceKonde** [comment]
Sorry for late response - use the .read() method of the Comparator objects; should return true or false. Didn't you read the documentation for the library it's right.... uhh, errrr... wrf. Missing from the README.md for the library, that's an easy fix will be in 2.4.0 later today. 

If that doesn't work that is a bug, open an issue.

**Reply by stevecolt** [reply]
Thanks for that, I'm glad that for a change it wasn't me just being dumb!! I think its working, but Ive managed to break the serial connection on the curiosity> I thought this was going to be a more solid interface than the serial. Frankly I'm suspicious of any solution that involves resistors and versions of software to fix. I don't get what a 4k7 resistor in series does, other than limit current to 1mA and make the whole thing more susceptible to stray capacitance, which if its a speed problem is much more likely to be an issue. Anyway will sort and report whats happening.
Thanks
Steve

**Reply by SpenceKonde** [reply]
Not sure what you're talking about, maybe start a new topic or issue (if it's a defect in core or documentation that needs to be fixed).

**Comment by stevecolt** [comment]
Sorry, off topic rant. Had spent ages trying to get serial programming
working, so went to curiosity, and now that's stopped working on serial!!
Just an observation on serial programming. Will  let you know how it goes
with the library.
Thanks
Steve

**Reply by stevecolt** [reply]
Just to say that Comparator.read() works fine at least on 0 & 1.
Thanks for your help
Steve

**Comment by SpenceKonde** [comment]
I would be truly in awe if it didn't (it's 1 line - and a very short line too, something like return !!(AC.STATUS&AC_STATE_bm), AC being the member variable that's a reference AC0 (or AC1/AC2 on parts that have them). :-P Glad it's sorted.
